src/Types.elm

```diff
@@ -71,6 +71,7 @@ listToTypeSign types =
 typeSignToList : VType -> List VType
 typeSignToList vt =
   case vt of
-    TBool -> [TBool]
-    TInt -> [TInt]
-    TFun vt1 vt2 -> (typeSignToList vt1) ++ (typeSignToList vt2)
+    TBool           -> [TBool]
+    TInt            -> [TInt]
+    TFun vt1 vt2    -> (typeSignToList vt1) ++ (typeSignToList vt2)
+    TTuple vt1 vt2  -> (typeSignToList vt1) ++ (typeSignToList vt2)   -- Will return the tuple as a list. Not ideal, but the function is type sign to list
```

Comment:
I think `typeSignToList (TTuple vt1 vt2 )` should return `[TTuple vt1 vt2 ]`.

The point of the typeSignToList function was to get the individual types into list form so that they're easier to work with. So if we have `TFun (TInt) (TFun TInt TBool)` this would be converted to `[TInt, TInt, TBool]` so that we can easily  work with the components of the type signature (for example, having this vtype in list form makes it easier print this function type as `TInt -> TInt -> TBool`).

Having said that I think it would make more sense to have this case return a list containing `[VTuple vt1 vt2]`. Ultimately we'll have to refactor the way we're obtaining the type signature so I don't think it's a huge deal but for now we should probably change it.